### ISSUE TITLE
Junit test failures on OS X and  Windows 7

### ISSUE THREAD

**Issue by bermud** [opengeospatial/ets-gml32#8]
When building r17 and r18

Failed tests: 
  VerifyPropertyValueTests.locationName_unknownHost 
Expected: (exception with message a string containing "Failed to connect to URL" and an instance of java.lang.AssertionError)
     got: <java.lang.AssertionError: Response entity is missing. expected [true] but found [false]>

  VerifyETSAssert.assertURLIsResolvable_unknownHost 
Expected: (exception with message a string containing "Failed to connect to URL" and an instance of java.lang.AssertionError)
     got: <java.lang.AssertionError: Response entity is missing. expected [true] but found [false]>

**Comment by rjmartell** [comment]
Luis,

Could you please add the part of the diagnostic message following "got:"? Or was that everything?

**Comment by bermud** [comment]
@rjmartell, 

- I think there was more after "got:" I will try again and provide the complete message
- java version "1.7.0_45"
- OS X 10.9.5

**Comment by bermud** [comment]
I tried to build master. Here is the error:

```
Failed tests: 
  VerifyETSAssert.assertURLIsResolvable_unknownHost 
Expected: (an instance of java.lang.AssertionError and exception with message a string containing "Failed to connect to URL")
     but: exception with message a string containing "Failed to connect to URL" message was "Response entity is missing. expected [true] but found [false]"
Stacktrace was: java.lang.AssertionError: Response entity is missing. expected [true] but found [false]
	at org.testng.Assert.fail(Assert.java:94)
	at org.testng.Assert.failNotEquals(Assert.java:494)
	at org.testng.Assert.assertTrue(Assert.java:42)
	at org.opengis.cite.iso19136.ETSAssert.assertURLIsResolvable(ETSAssert.java:187)
	at org.opengis.cite.iso19136.VerifyETSAssert.assertURLIsResolvable_unknownHost(VerifyETSAssert.java:99)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:239)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:283)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:173)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:128)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)

  VerifyPropertyValueTests.locationName_unknownHost 
Expected: (an instance of java.lang.AssertionError and exception with message a string containing "Failed to connect to URL")
     but: exception with message a string containing "Failed to connect to URL" message was "Response entity is missing. expected [true] but found [false]"
Stacktrace was: java.lang.AssertionError: Response entity is missing. expected [true] but found [false]
	at org.testng.Assert.fail(Assert.java:94)
	at org.testng.Assert.failNotEquals(Assert.java:494)
	at org.testng.Assert.assertTrue(Assert.java:42)
	at org.opengis.cite.iso19136.ETSAssert.assertURLIsResolvable(ETSAssert.java:187)
	at org.opengis.cite.iso19136.data.PropertyValueTests.validateLocationName(PropertyValueTests.java:75)
	at org.opengis.cite.iso19136.data.VerifyPropertyValueTests.locationName_unknownHost(VerifyPropertyValueTests.java:75)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:239)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:283)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:173)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:128)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)




Tests run: 130, Failures: 2, Errors: 0, Skipped: 4

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 18.548 s
[INFO] Finished at: 2015-02-24T13:38:09-05:00
[INFO] Final Memory: 25M/447M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.18:test (default-test) on project ets-gml32: There are test failures.
[ERROR] 
[ERROR] Please refer to /Users/lbermudez/Documents/Dropbox/github/ets-gml32/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

**Comment by bermud** [comment]
@rjmartell, have you checked on this further?
if skipping tests, it works fine: 
```mvn install -Dmaven.test.skip=true```

**Comment by bermud** [comment]
fixed in 71def06af39f8b491a6d1033254641ef0ee214e5

**Comment by rjmartell** [comment]
This appears to be a platform-specific issue. For example, it does not occur when building on Windows 7 or a recent Linux distribution using the Oracle JDK 7. The unit test should fail with an `IOException` because the URI http://codespace.nowhere.info/ is not accessible.

**Comment by rjmartell** [comment]
I was able to build the test suite on OS X 10.7.4 using Oracle JDK 7u79.

**Comment by bermud** [comment]
I'm opening this issue again, but changed the title to be OS specific.

**Comment by rjmartell** [comment]
Exactly what configuration is affected? For example, it builds on an older MacBook (2010) running OS X 10.7.4 with Oracle JDK 7u79.

**Comment by sMorrone** [comment]
When building r1.21 on Windows 7,  java version 1.8.0_45, following tests fail:

VerifyPropertyValueTests.locationName_unknownHost 
![screenshot 2015-07-09 15 01 29](https://cloud.githubusercontent.com/assets/13329248/8669478/59aa8122-2a15-11e5-9474-8eef8c67224a.png)

VerifyETSAssert.assertURLIsResolvable_unknownHost 
![screenshot 2015-07-09 14 43 58](https://cloud.githubusercontent.com/assets/13329248/8669449/118b683e-2a15-11e5-8086-37428cdf5e4e.png)

I skipped the tests, so I was able to build, but following considerations arise:
1) the issue is not  'OS specific' 
2) exactly what am I missing skipping these tests?

**Comment by bermud** [comment]
Hi @sMorrone thank you for reporting the issue. I updated the title. This one has been hard to debug. We have tested in other OS machines and it works fine. The tests ensure that any changes to the code are not breaking anything. It is mostly used by the lead developer. But we sure need to find the problem.

@rjmartell, what do you suggest?

**Comment by rjmartell** [comment]
The original report was about building the ETS with JDK 7. It builds successfully using Oracle JDK 7u79 (x64) on Windows 7 Pro SP1. 

`Tests run: 134, Failures: 0, Errors: 0, Skipped: 3`

I'll try building it with the latest JDK 8.

**Comment by rjmartell** [comment]
All unit tests passed using Oracle JDK 8u51 (x64) on Windows 7 Pro SP1. However, the build failed due to Javadoc errors (reported with doclint enabled). I cleaned these up on the master branch.

**Comment by rjmartell** [comment]
The test suite builds cleanly in the following environments:

Operating System | JDK | Maven
----- | ----- | -----
Windows 7 Pro SP1 (x64) | Oracle 7u79, 8u51 | 3.3.3
RHEL 6.7 (x64) | Oracle 7u79, 8u51 | 3.3.3

**Comment by rjmartell** [comment]
Fixed by commit 0a7c0685466a94758e18f18a0235f1ad64d3a8e8
